### DISCUSSION TITLE
[백준 1932] 정수 삼각형

### DISCUSSION THREAD

**Discussion by BanApp** [ghdcksgml1/Algorithm_Study#91] [1일 1알고리즘]
<img width="598" alt="스크린샷 2022-02-28 오후 4 31 25" src="https://user-images.githubusercontent.com/93313445/155942092-ffd5a615-fa77-456a-a62b-de542f4d1f7d.png">

```
pypy3

import sys
n = int(sys.stdin.readline())

arr = []
for i in range(n):
    arr.append(list(map(int, sys.stdin.readline().split())))

for i in range(1,n):
    for j in range(i+1):
        if j == 0:
            arr[i][j] = arr[i][j] + arr[i-1][j]
        elif i == j:
            arr[i][j] = arr[i][j] + arr[i-1][j-1]
        else:
            arr[i][j] = max(arr[i][j]+arr[i-1][j], arr[i][j]+arr[i-1][j-1])

print(max(arr[n-1]))
```

언제 풀어도 즐거운 DP 문제!
이 문제의 핵심은 연산된 결과 중 더 큰 값을 기존 리스트에 존재하는 값으로 대체해서 누적시키는 것이다.
양 쪽 끝은 올 수 있는 경우의 수가 한 가지씩 밖에 없다.
왼쪽 끝은 바로 한 칸 위 제일 왼쪽, 오른쪽 끝은 바로 한 칸 위 현재 위치부터 한 칸 왼쪽을 더해주면 되고 그 사이에 있는 애들은 두 가지 경우 중에 더 큰애들을 넣어주도록 max 함수를 사용한다.
그렇게 되면 n-1 번째 줄에 더해서 나올 수 있는 모든 경우가 다 담기는데 이것들 중에서 최댓값을 출력해 주면 정답이다!

**Comment by ghdcksgml1** [comment]
ㅋㅋㅋㅋㅋ dp가 즐겁다니...ㅋㅋㅋㅋㅋㅋㅋ

**Reply by BanApp** [reply]
실버까지는 즐거움.
